pkg/query/flamegraph_arrow.go

```diff
@@ -1788,11 +1811,17 @@ func (fb *flamegraphBuilder) trim(ctx context.Context, tracer trace.Tracer, thre
 
 		cumThreshold := float32(cum) * threshold
 
+		valueOffset := te.valueOffset
 		for _, cr := range fb.childrenList[te.row] {
 			if v := fb.builderCumulative.Value(cr); v > int64(cumThreshold) {
 				// this row is above the threshold, so we need to keep it
 				// add this row to the queue to check its children.
-				trimmingQueue.push(trimmingElement{row: cr, parent: row})
+				trimmingQueue.push(trimmingElement{
+					row:         cr,
+					parent:      row,
+					valueOffset: valueOffset,
+				})
+				valueOffset += fb.builderCumulative.Value(cr)
```

Comment:
Ok. So the valueOffset is the offset from the other children relative to the parent's cumulative value, right? 
I just wanted to verify that we're not passing the pixel from the frontend and performing some x-y axis math in the backend. 
This makes sense to me.

Reply:
correct it's the value relative to the total, the frontend can then use that to calculate the position based on the available pixel space